### PR TITLE
Fix generating the MSVC 2005/2008 project

### PR DESCRIPTION
**Pull request by ccawley2011**

### CODE REVIEW

**Comment by sezero** [comment]
You can remove the `TargetFrameworkVersion` lines too, now that you went back from vs2008 to vs2005

**Comment by sezero** [comment]
Also would be a good idea to add x64 target to the solution.

PS: if #431 gets merged, this will need rebasing + adjusting.

**Comment by ccawley2011** [comment]
> Also would be a good idea to add x64 target to the solution.

x64 support isn't provided by the Express editions prior to VS2012, so I wouldn't be able to test this.

The other points have been addressed.

File: test-dev/main.c [hunk 1]
```diff
@@ -6,6 +6,8 @@
 #ifndef WIN32_LEAN_AND_MEAN
 #define WIN32_LEAN_AND_MEAN
 #endif
+#undef _WIN32_WINNT
+#define _WIN32_WINNT 0x0400
```
Comment (sezero) [inline test-dev/main.c:10]: IMO: This second patch actually deserves applying independent of the vc project patch.

One thing of note: do we actually want to force `_WIN32_WINNT` to `0x0400`?
Wouldn't it be more appropriate if we set it to `0x0400` if it's not already set?
(Unless I'm missing something.)

**Comment by sezero** [comment]
How does the first vc project file patch work with the `lite` version?

File: src/filter.c [hunk 2]
```diff
@@ -68,6 +68,7 @@ static const float resonance_table[128] = {
 
 
 #if !defined(HAVE_POWF) || defined(__DJGPP__) || defined(__WATCOMC__)
+#undef powf
```
Comment (sezero) [inline src/filter.c:71]: You can move this to the second patch

**Comment by sezero** [comment]
Here is a patch, on top of yours, that does the following:
- adds x64 target
- disables debug info generation - the proj is 'Release' only anyway
- disables pdb generation - same reason as above
- disables c++ exceptions option - this is C-only
- changes subsystem to 'console'
[patch.txt](https://github.com/libxmp/libxmp/files/6842325/patch.txt)

**Comment by sezero** [comment]
Another thing I wonder: why did @cmatsuoka disabled this one?
Not as flexible as Makefile.vc?

Speaking of flexibility, the project-file method is pretty rigid in its way
about prowizard and depackers enable/disable option.

@AliceLR, @cmatsuoka: What do you think of this PR?

File: vc/libxmp.vcproj [hunk 3]
```diff
@@ -1,7 +1,7 @@
 ﻿<?xml version="1.0" encoding="UTF-8"?>
 <VisualStudioProject
 	ProjectType="Visual C++"
-	Version="8.00"
+	Version="8,00"
```
Comment (sezero) [inline vc/libxmp.vcproj:4]: Apologies -- this `.` to `,` change seems to be because of my windows regional setting and is not necessary.

**Comment by ccawley2011** [comment]
I've also been testing with PR #415, and it does work with VC 2005 with some modifications, with the one caveat being that it requires CMake 3.11 or older. In the long run, it might be a better option, but as long as the old solution is still here, it's worth fixing.

**Comment by sezero** [comment]
>  one caveat being that it requires CMake 3.11

Hm? It requires 3.2 in the cmake script, I briefly tested today using 3.9.6

> or older.

Hm?? `Newer`, right?

>  as long as the old solution is still here, it's worth fixing.

That I agree myself.

**Comment by ccawley2011** [comment]
> > one caveat being that it requires CMake 3.11
> 
> Hm? It requires 3.2 in the cmake script, I briefly tested today using 3.9.6
> 
> > or older.
> 
> Hm?? `Newer`, right?

I'm specifically referring to the requirements for generating Visual Studio 2005 projects: https://cmake.org/cmake/help/v3.21/generator/Visual%20Studio%208%202005.html

**Comment by sezero** [comment]
> I'm specifically referring to the requirements for generating Visual Studio 2005 projects: https://cmake.org/cmake/help/v3.21/generator/Visual%20Studio%208%202005.html

Huh.. In that case, you are right.

**Comment by sezero** [comment]
This needs rebasing again: #433 is merged.
As for powf macro in filter.c: See #435: better?

**Comment by AliceLR** [comment]
> What do you think of this PR?

My usage of MSVC has been limited mostly to testing existing buildsystems targeted for MSVC so I don't think I have any particularly useful feedback re: this.

**Comment by sezero** [comment]
OK, having this actually does not hurt - options are good IMO.
To summarize:

- `make dist` (full library) includes the vc directory - OK: good,

- `lite` version does not include this, and does not generate a
   'lite' project file - Not OK: we should have this, IMO.

- generation of project file does not respect prowizaro/depackers
  disable options: - I'd say not OK, but I also don't know whether
  that is doable.

Comments? Cameron?

**Comment by sezero** [comment]
Additional question:

- We already have `Makefile.vc` to support builds using MSVC.
  How advantageous is having an additional project file?

**Comment by sezero** [comment]
I applied almost all parts:
https://github.com/libxmp/libxmp/commit/34af5adc873955a0ade1f2cce99a1443f747ee60
https://github.com/libxmp/libxmp/commit/966cfa5b3bd7e93e343adbf51da92da97b5ea780
https://github.com/libxmp/libxmp/commit/911ee36e9e21198c9a72a1992bf46f90d0fb7413
https://github.com/libxmp/libxmp/commit/b679a79b7a79ff914d70a45c357bcbb06336e241

The missing parts are: (1) integration into main tree, and (2) generated project file.

Will try working on this more.

**Comment by sezero** [comment]
If I change vc makefile like this:
```diff
--- vc/Makefile~
+++ vc/Makefile
@@ -18 +18 @@
-	@for i in $(OBJS:.o=.c) $(PROWIZARD_OBJS:.o=.c) $(DEPACKER_OBJS:.o=.c); do \
+	@for i in $(ALL_OBJS:.o=.c); do \
```
then it does obey the `--disable-depackers` and `--disable-prowizard` configure switches,
but only as far as source list is concerned. Still not good enough (I think)..

**Comment by sezero** [comment]
This looks like a WONTFIX, yes?

**Pull request closed without merging**